controller/pokemon_test.go

```diff
@@ -0,0 +1,231 @@
+	"net/url"
+)
+
+var getPokemonCSVMock func() (pokemon model.PokemonList, err error)
+var getPokemonCSVByIdMock func() (pokemon model.Pokemon, err error)
+var getPokemonThroughApiMock func() (pokemon model.FetchedPokemon, err error)
+var getPokemonMock func() (pokemon model.PokemonList, err error)
+var getPokemonByIdMock func() (pokemon model.Pokemon, err error)
+var getPokemonFromReaderMock func() (pokemon model.Pokemon, err error)
+var addPokemonMock func() (pokemon model.Pokemon, err error)
+var updatePokemonMock func() (pokemon model.Pokemon, err error)
+var deletePokemonMock func() (pokemon model.Pokemon, err error)
+
+var r, _ = http.NewRequest("GET", "http://example.com", nil)
+var pokemon = model.Pokemon{}
+var pokemonList = []model.Pokemon{pokemon, pokemon}
+
+type PokemonControllerMock struct{}
+
+func (pc PokemonControllerMock) getPokemonCSV(queryParams url.Values) (model.PokemonList, error) {
+	return getPokemonCSVMock()
+}
+
+func (pc PokemonControllerMock) getPokemonCSVById(id string) (model.Pokemon, error) {
+	return getPokemonCSVByIdMock()
+}
+
+func (pc PokemonControllerMock) getPokemonThroughApi(name string) (model.FetchedPokemon, error) {
+	return getPokemonThroughApiMock()
+}
+
+func (pc PokemonControllerMock) getPokemon() (model.PokemonList, error) {
+	return getPokemonMock()
+}
+
+func (pc PokemonControllerMock) getPokemonById(id string) (model.Pokemon, error) {
+	return getPokemonByIdMock()
+}
+
+func (pc PokemonControllerMock) getPokemonFromReader(reader io.ReadCloser) (model.Pokemon, error) {
+	return getPokemonFromReaderMock()
+}
+
+func (pc PokemonControllerMock) addPokemon(pokemon model.Pokemon) (model.Pokemon, error) {
+	return addPokemonMock()
+}
+
+func (pc PokemonControllerMock) updatePokemon(id string, pokemon model.Pokemon) (model.Pokemon, error) {
+	return updatePokemonMock()
+}
+
+func (pc PokemonControllerMock) deletePokemon(id string) (model.Pokemon, error) {
+	return deletePokemonMock()
+}
+
+func printResponse(w *httptest.ResponseRecorder) {
+	res := w.Result()
+	body, _ := io.ReadAll(res.Body)
+	bodyAsString := string(body)
+
+	fmt.Println(res.StatusCode)
+	fmt.Println(res.Header.Get("Content-Type"))
+	fmt.Println(bodyAsString)
+}
+
+func ExampleGetPokemonCSV() {
+	pc = PokemonControllerMock{}
+	w := httptest.NewRecorder()
+
+	getPokemonCSVMock = func() (pokemon model.PokemonList, err error) {
+		return pokemonList, nil
+	}
+
+	GetPokemonCSV(w, r)
+	printResponse(w)
+
+	w = httptest.NewRecorder()
+	var onePokemonList = []model.Pokemon{pokemon}
+
+	getPokemonCSVMock = func() (pokemon model.PokemonList, err error) {
+		return onePokemonList, nil
+	}
+
+	GetPokemonCSV(w, r)
+	printResponse(w)
+
+	// Output:
+	// 200
+	// application/json
+	// [{"id":0,"pokedexNumber":0,"name":"","types":"","region":""},{"id":0,"pokedexNumber":0,"name":"","types":"","region":""}]
+	//
+	// 200
+	// application/json
+	// {"id":0,"pokedexNumber":0,"name":"","types":"","region":""}
+}
+
+func ExampleGetPokemonCSVById() {
```

Comment:
Your test functions should be named like `TestGetPokemon`, I believe this is how Go executes the test functions

Reply:
Yes but there are also Example functions that checks the output of a fmt statement